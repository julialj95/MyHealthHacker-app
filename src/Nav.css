.nav {
  flex-grow: 2;
}
.navbar {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
}

.navlink {
  list-style: none;
}
