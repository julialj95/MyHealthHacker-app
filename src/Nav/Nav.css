@import url("https://fonts.googleapis.com/css2?family=Lato&family=Nunito&family=Source+Sans+Pro&display=swap");
.nav {
  flex-grow: 2;
}
.navbar {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  text-align: left;
}

.navlink {
  list-style-type: none;
}
.link_style {
  color: white;
  text-decoration: none;
  font-family: "Nunito";
  font-size: 14px;
}

.logout_btn_container {
  list-style-type: none;
  text-align: left;
}

.logout_btn {
  background: none;
  border: none;
  padding: 0;
  cursor: pointer;
}

@media all and (min-width: 600px) {
  .navbar {
    flex-direction: row;
    justify-content: space-evenly;
  }
  .link_style {
    font-size: 16px;
  }
}
@media all and (min-width: 900px) {
  .link_style {
    font-size: 19px;
  }
}
