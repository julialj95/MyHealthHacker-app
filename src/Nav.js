import React from "react";
import { Link } from "react-router-dom";
import "./Nav.css";

class Nav extends React.Component {
  render() {
    return (
      <nav className="nav">
        <ul className="navbar">
          <li>
            <Link className="navlink" to="/">
              Home
            </Link>
          </li>
          <li>
            <Link className="navlink" to="/submit-log">
              Submit Log
            </Link>
          </li>
          <li>
            <Link className="navlink" to="/logs">
              View Logs
            </Link>
          </li>
          <li>
            <Link className="navlink" to="signup">
              Create Account
            </Link>
          </li>
          <li>
            <Link className="navlink" to="login">
              Log In
            </Link>
          </li>
        </ul>
      </nav>
    );
  }
}

export default Nav;
