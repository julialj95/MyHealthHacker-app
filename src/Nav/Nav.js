import React, { useContext } from "react";
import { Link } from "react-router-dom";
import "./Nav.css";
import UserContext from "../UserContext";

function Nav(props) {
  const context = useContext(UserContext);
  console.log("context", context);
  const LoggedIn = () => {
    return (
      <>
        <li className="navlink">
          <Link to="/submit-log">Submit Log</Link>
        </li>
        <li className="navlink">
          <Link to="/logs">View Logs</Link>
        </li>
        <li>
          <button className="logout_btn" onClick={props.handleLogout}>
            Log Out {context.username}
          </button>
        </li>
      </>
    );
  };

  const LoggedOut = () => {
    return (
      <>
        <li className="navlink">
          <Link to="signup">Create Account</Link>
        </li>
        <li className="navlink">
          <Link to="login">Log In</Link>
        </li>
      </>
    );
  };
  console.log("props.loggedIn", props.loggedIn);
  return (
    <nav className="nav">
      <ul className="navbar">
        <li className="navlink">
          <Link to="/">Home</Link>
        </li>
        {props.loggedIn ? LoggedIn() : LoggedOut()}
      </ul>
    </nav>
  );
}

export default Nav;
