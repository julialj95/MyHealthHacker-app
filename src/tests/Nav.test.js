import React from "react";
import ReactDOM from "react-dom";
import Nav from "../Nav/Nav";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");

  ReactDOM.render(
    <Router>
      <Nav />
    </Router>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <Router>
        <Nav />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
