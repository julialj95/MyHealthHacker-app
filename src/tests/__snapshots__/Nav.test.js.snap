// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<nav
  className="nav"
>
  <ul
    className="navbar"
  >
    <li
      className="navlink"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="navlink"
    >
      <a
        href="/signup"
        onClick={[Function]}
      >
        Create Account
      </a>
    </li>
    <li
      className="navlink"
    >
      <a
        href="/login"
        onClick={[Function]}
      >
        Log In
      </a>
    </li>
  </ul>
</nav>
`;
